packages/edit-site/src/components/page-templates/index.js

```diff
@@ -315,6 +329,50 @@ export default function PageTemplates() {
 				elements: authors,
 				width: '1%',
 			},
+			{
+				header: __( 'Post types' ),
+				id: 'postTypes',
+				getValue: ( { item } ) =>
```

Comment:
This `getValue` function is used by the filters. Note how the values returned here are different from the values returned by `render`, which is not what we've been doing so far.

This enables us to implement the same [logic as the server](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/block-template-utils.php#L1077). For example, when querying the templates endpoint with `post_type: post` the server returns:

1. Templates whose `templates->post_types` values contain `post`.
2. User templates that are custom but don't have any post type information.

Additionally, there's a few default templates with special meaning that are not returned by the endpoint in that scenario because they're the default and we should probably display here when the user filters by post type:

3. The `single` & `singular` templates, as per the [template hierarchy](https://developer.wordpress.org/themes/basics/template-hierarchy/#visual-overview).

Reply:
An alternative to this approach would be allowing the field to provide a custom filter callback. I'm investigating this next, though I wanted to share as early as possible in case anyone has different ideas.

Reply:
I've extracted a `getPostTypesFromItem` method that centralizes all this logic. I feel better about this now. Still need to review/add support for CPTs to assess whether this is a good enough approach.